recursive_art.py

```diff
@@ -1,99 +1,62 @@
 """ This program generates random computational art. The input argument is the file name.
     The default size of the image is 350x350 pixels. The RGB values of each pixel are determined
-    by building a random function for each color and then evaluate those funcitons at each
-    pixel. By design, the functions have a depth of 7-9 function levels but the program
-    could easily be altered to create deeper or shallower functions.
+    by building a random function for each color using lambda functions and then recursively
+    nesting those lambda functions within each other to create a single function for each color.
+    The functions are then given the pixel coordiantes as x,y inputs and evaluated at each point.
+    This is a going beyond extension of the original computational art project which can be found
+    in the release branch of this repo.
 
     Author : Lakhvinder Jordan <ljordan51@gmail.com>
     Course : Olin Software Design Spring 2017
-    Date   : 2017-02-21
+    Date   : 2017-04-21
 """
 
 import random
 import math
 from PIL import Image
 
+# setting up lambda functions for build_func to use
+prod = lambda x, y: x*y
```

Comment:
One of the benefits of lambda functions is that they can be defined anonymously, or inline - for example, without making these "flags" for them. However, this is still a quick/short way to define operations and pass them around, so that's acceptable - in fact, it makes it significantly more readable than randomly selecting nameless lambda functions. 

Plus, the execution is way cleaner here than with the giant list of if/else statements - a well-scoped extension, well-executed.